### PR TITLE
Peer reviews

### PR DESCRIPTION
**Pull request by gnott**
Re eLife assessment issue https://github.com/elifesciences/issues/issues/8030
Re public reviews issue https://github.com/elifesciences/issues/issues/8031
Re epic issue https://github.com/elifesciences/issues/issues/8029

eLife is adding more types of peer review content in `<sub-article>` XML tags. For some of them the `@article-type` attribute overlaps with previously configured peer review parsing functions and they've been refactored in a mostly-backwards-compatible way.

Determining which `<sub-article>` is which now depends on looking for its `<article-title>` tag value and to match whether a term is present, or exclude it if the term is present.

To @lsh-0 please be welcome to review as you please, although I may release this in my next work day if you don't get a chance. It works, and I'm not aiming for perfection, since the test cases give me confidence it will probably satisfy the immediate needs.

### CODE REVIEW

**Comment by lsh-0** [comment]
> To @lsh-0 please be welcome to review as you please, although I may release this in my next work day if you don't get a chance. It works, and I'm not aiming for perfection, since the test cases give me confident it will probably satisfy the immediate needs.

cheers, @gnott , I'll take a look today, thanks for making it available. Everything downstream of the api-raml is *much* more malleable than the specification so even if we miss something here it's relatively trivial to fix it when we detect it later.

I appreciate this is like three big tickets in one

**Comment by gnott** [comment]
Thanks, actually also go ahead and make changes directly and push them if you want, as long as the tests pass I'm happy.

File: tests/fixtures/test_recommendations_for_authors/content_01_expected.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from collections import OrderedDict
+
+expected = OrderedDict()
```
Comment (lsh-0) [inline tests/fixtures/test_recommendations_for_authors/content_01_expected.py:3]: just a curious question, no need to change anything, but in these cases with empty expectations what is `content_01` vs `content_02`? is it the old kitchen sink vs the new one? 

and in this specific case, if this fixture is being used for `test_recommendations_for_authors` why are we testing against an `editorReport`? 

From this:
https://github.com/elifesciences/XML-mapping/blob/a559aaa70ac6239528e9ecabe0d2e69802337a95/elife-1234567890-v2/elife-1234567890-v2.xml#L2005

it does look like an "editor's report" changed to an "editor's evaluation" that then changed to an "elife assessment" ... actually I just took a look at the other fixture and it looks like a copy+paste job so just ignore me ;)

**Review by lsh-0 — APPROVED** [review]
looks good

File: tests/fixtures/test_recommendations_for_authors/content_01_expected.py (hunk 1 above)
Comment (gnott) [reply tests/fixtures/test_recommendations_for_authors/content_01_expected.py:3]: I wasn't too picky about the XML fixtures in some of these test cases. The old kitchen sink `content_01` test here is an example of an XML which has a `<sub-article>` tag and will not return `Recommendations for authors` content.

It is probably a bit of overkill in this case, but it is slightly different than parsing `<root/>` (which doesn't contain any `<sub-article>` tags), and the `content_02` test is an example which will return a `Recommendations for authors`. I threw in some test cases on the old kitchen sink style XML to test for empty return values and guard against regressions.

**Pull request merged**